### ISSUE TITLE
Investigate options for IDE integration for C++ projects

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#19]
This story is to research and perhaps spike some short-term and long-term options for IDE integration. The output should be some design document from which we can plan further work. The actual work on the integrations will be later stories.

**Comment by lacasseio** [comment]
I'm moving this in the ice box as we decided to move with Xcode for the time been. We can investigate at a later time.

**Comment by lacasseio** [comment]
I rescope this issue into a future epic to support cross-platform IDE integration.

**Comment by adammurdoch** [comment]
Let's change the scope back to just "ide integration". The new title suggests that we'll only consider IDE integrations that are cross-platform, which is not the case. We will also consider an IDE integration that is specific to a single platform but also extremely popular (visual studio, for example).

**Comment by lacasseio** [comment]
Good point @adammurdoch. I was coming from the angle that we currently have Xcode integration for both Swift and C++ and the current gap is a cross-platform solution. We could probably scope this as an all major platform support for IDE integration. Xcode handles macOS, Visual Studio could be Windows, and I don't know what is popular on Linux.

**Comment by adammurdoch** [comment]
The scope of this issue should be something like "investigate the options to allow most developers to use an IDE to develop native software with Gradle". We might solve this problem by selecting a specific popular IDE for each popular platform, or we might solve it by selecting a single IDE that works on every popular platform, or some combination.

At this point, until we've done the investigation and written up the results, we shouldn't speculate what that solution will be.
